algorithms/map-optimization/src/outlier-rejection-solver.cc

```diff
@@ -157,17 +159,33 @@ ceres::TerminationType solveStep(
 
   // Disable the default Ceres output.
   local_options.minimizer_progress_to_stdout = false;
-  local_options.max_num_iterations = num_iters;
   local_options.update_state_every_iteration = true;
-
-  ceres::Solver::Summary summary;
-  ceres::Solve(local_options, &problem, &summary);
-
-  // Save the solver summary of each iteration.
-  if (result != nullptr) {
-    result->solver_summaries.emplace_back(summary);
+  if (max_solver_time_s) {
+    double remaining_time = *max_solver_time_s;
+    local_options.max_num_iterations = 1;
```

Comment:
have you noticed any difference in doing a manual loop around ceres in comparison to doing 3 inner loops? I assume the problem is kept allocated